src/ngx_http_upsync_module.c

```diff
@@ -1353,6 +1366,148 @@ ngx_http_upsync_consul_parse_json(void *data)
 
 
 static ngx_int_t
+ngx_http_upsync_consul_services_parse_json(void *data)
+{
+    ngx_buf_t                      *buf;
+    ngx_http_upsync_ctx_t          *ctx;
+    ngx_http_upsync_conf_t         *upstream_conf = NULL;
+    ngx_http_upsync_server_t       *upsync_server = data;
+    ngx_int_t                       attr_value;
+
+    ctx = &upsync_server->ctx;
+    buf = &ctx->body;
+
+    cJSON *root = cJSON_Parse((char *)buf->pos);
+    if (root == NULL) {
+        ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, 0,
+                      "upsync_parse_json: root error");
+        return NGX_ERROR;
+    }
+
+    if (ngx_array_init(&ctx->upstream_conf, ctx->pool, 16,
+                       sizeof(*upstream_conf)) != NGX_OK)
+    {
```

Comment:
In function *ngx_http_upsync_consul_parse_json()* it's put this way: actually I copied this code from this function. So do you really want that I change that?